src/PaymentGateways/Gateways/Stripe/CheckoutGateway.php

```diff
@@ -111,11 +111,12 @@ public function getLegacyFormFieldMarkup(int $formId, array $args): string
         Stripe::canShowBillingAddress($formId, $args);
 
         switch (give_stripe_get_checkout_type()) {
-            case 'modal':
-                return $this->getCheckoutInstructions()
-                    . $this->getCheckoutModalHTML($formId, $args);
             case 'redirect':
                 return $this->getCheckoutInstructions();
+            case 'modal':
+            default:
+                return $this->getCheckoutInstructions()
+                       . $this->getCheckoutModalHTML($formId, $args);
```

Comment:
Is this intended to handle any other value? If not, default should throw an exception.

Reply:
Just the two values, with `modal` being the default value.

I added an exception, see [c25fe0b](https://github.com/impress-org/givewp/pull/6513/commits/c25fe0b243f0febae3159b9b81ac0e7e0f4ef9a0).